### PR TITLE
[Frontend][Driver] Remove `compile_from_mlir`.

### PR DESCRIPTION
**Pull request by erick-xanadu**
**Context:** In order to make the compiler driver a standalone executable, we would to simplify the compiler driver interface. I.e., the compiler driver executable should just produce a shared object. The `struct FunctionAttributes` which contained two string fields (one for the name of the entry point of the function and the other for the return values of the entry point) was only used when lowering MLIR from the textual representation, which has been useful for debugging. 

**Description of the Change:** Remove `compile_from_mlir`.

**Benefits:** The benefit is that the compiler driver can just focus on compiling without having to return extra metadata which simplifies the interface between the compiler driver and the python process running it.

**Possible Drawbacks:** None, use `replace_ir` instead.

**Related GitHub Issues:**

EDIT: I do believe I can keep it being pythonic by using the following syntactic sugar of the docstring. We can use the python's function signature to derive the signature of the MLIR function (just like we do right now for the normal lowering) and we use the same name for the QJIT-ed function in python as the entry_point (similar to what we do in QJIT). During compilation, we can see the bytecode and find out that the function only contains a docstring and returns None, which would then allow us to decide to compile the docstring.

```python
@qjit
def identity(x: float) -> float:
  """
  func.func identity(tensor<f64> %arg) -> tensor<f64> {
    return %arg: tensor<f64>
  }
  """
```

### CODE REVIEW

**Comment by mehrdad2m** [comment]
Thank you @erick-xanadu, This is great! 🚀

**Review by mehrdad2m — APPROVED** [review]

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -218,6 +218,13 @@
   This is unlikely to affect users since only under certain conditions did
   nesting qnodes worked successfully.
 
+* Changes in the `debug.compile_from_mlir`'s API.
```
Comment (dime10) [inline doc/releases/changelog-dev.md:221]: Wait is this function even still needed? I thought it was supplanted by `replace_ir`

Comment (erick-xanadu) [reply doc/releases/changelog-dev.md:221]: Let me look and see if anyone is using it. If not, I will also delete it. :)

Comment (erick-xanadu) [reply doc/releases/changelog-dev.md]: https://github.com/PennyLaneAI/catalyst/pull/1181/commits/2e723e0b6e2f955503ecb8a963ef1293e9337378

Comment (dime10) [reply doc/releases/changelog-dev.md]: Love it!

**Review by dime10 — APPROVED** [review]
💯

**Pull request merged**